lib/CAPI/CMakeLists.txt

```diff
@@ -11,9 +11,37 @@ add_mlir_public_c_api_library(TorchMLIRCAPI
   LINK_COMPONENTS
   Core
 
+  DEPENDS
+  MLIRAffinePassIncGen
+  MLIRArithmeticTransformsIncGen
+  MLIRBufferizationPassIncGen
+  MLIRConversionPassIncGen
+  MLIRFuncTransformsIncGen
+  MLIRLinalgPassIncGen
+  MLIRSCFPassIncGen
+  MLIRTensorTransformsIncGen
+
   LINK_LIBS PUBLIC
```

Comment:
For the dialect headers you include in the corresponding C file, it would be better to include the dialect libraries here (versus counting on transitively getting them from one of these deps)